app/controllers/redirections_controller.rb

```diff
@@ -1,4 +1,6 @@
 class RedirectionsController < ApplicationController
+  before_action :ensure_referrer_is_not_localhost
```

Comment:
An alternative to the before action would be to:

* Make a validator that made sure the host wasn't localhost
* Raise a validation error if it was
* `rescue_from` that error in the app controller and redirect to localhost page

I've used this pattern in APIs to great effect, and mostly avoided before_actinos. Do you have a preference?